### ISSUE TITLE
wxScrolledWindow runs Fit instead of FitInner

### ISSUE THREAD

**Issue by rexkogitans** [wxFormBuilder/wxFormBuilder#827]
When adding a wxScrolledWindow which has a sizer as child element, the sizer gets correctly associated with the scrolled window. However, the size is not correctly set.

This is a snippet of my latest project, this snippet is generated by wxFormBuilder:
```C++
	m_scrwinImportFormat->SetSizer( bSizer251 );
	m_scrwinImportFormat->Layout();
	bSizer251->Fit( m_scrwinImportFormat );
	m_notebookImport->AddPage( m_scrwinImportFormat, _("List format"), false );
```
**Instead of** `bSizer251->Fit( m_scrwinImportFormat );`, it should be `bSizer251->FitInside( m_scrwinImportFormat );`, otherwise the initial size of the sizer is measured against the parent element (in my case a wxNotebook) instead of the sizer's content and there will be no scrollbars.

I am looking forward to see this fixed.

Kind regards,
rexkogitans

**Comment by rexkogitans** [comment]
See this discussion: https://forums.wxwidgets.org/viewtopic.php?f=1&t=51416&p=224976#p224976

**Comment by rexkogitans** [comment]
Maybe I was not clear about the fact that the discussion on the wxWidgets forum I linked was also initiated by me. I am developing an app where I stumbled over this issue, and I played through several scenarios to fix this:

- In my child class, call `FitInside`. Since `Fit` is already invoked by the parent class which is generated by wxFormBuilder, `FitInside` does not have an effect.
- Modify the generated class (not recommended), by replacing `Fit` with `FitInside`. No need to add it in the child class, too. This results in the desired behaviour.
- My current workaround: Set the maximum size of the scrolled window to 0 by 0 pixels and leave everything as is: Do not touch the class generated by wxFormBuilder. Add the call to `FitInside` in my child class. This results in the desired behaviour.

Regarding your note:
    A note about the linked thread, i doubt that setting a maximum size leads to any change. Setting a non-default size however does skip the Fit-Calls.

This sounds like a contradiction. Indeed, it leads to a change: The `Fit`-call is skipped! That's why a subsequent call to `FitInside` is working.

**Comment by sodevel** [comment]
> Regarding your note: A note about the linked thread, i doubt that setting a maximum size leads to any change. Setting a non-default size however does skip the Fit-Calls.
> 
> This sounds like a contradiction. Indeed, it leads to a change: The `Fit`-call is skipped! That's why a subsequent call to `FitInside` is working.

No, changing `maximum_size` does not prevent the `Fit`-call, it is still there, you can see this easily in the code preview of wxFB. However, setting `size` to anything else but `wxDefaultSize` does omit the `Fit`-call.

The effect of specifying `maximum_size` is highly context dependend, i vaguely remember that many sizers don't resepect it at all. I tried to replicate your issue in a minimal example but failed. But anyway, i agree that the `Fit`-call should not be there, i just don't know yet how to do this in a sensible way.
